fanficfare/adapters/adapter_asianfanficscom.py

```diff
@@ -113,6 +113,23 @@ def doAdultCheck(self, url, soup):
         else:
             return False
 
+    def doSubCheck(self, url, soup):
+        check = soup.find('div',{'class':'click-to-read-full'})
+        if check:
+            logger.debug("Subscription required to get all HTML tags")
+            #does not work when using https - 403
+            subUrl = 'http://' + self.getSiteDomain() + soup.find('a',{'id':'subscribe'})['href']
+            self._fetchUrl(subUrl)
+            data = self._fetchUrl(url,usecache=False)
+            soup = self.make_soup(data)
+            check = soup.find('div',{'class':'click-to-read-full'})
+            if check:
+                raise exceptions.FailedToDownload("Error downloading Chapter: %s!  Missing required element!" % url)
+            else:
+                return soup
+        else:
+            return False
```

Comment:
Rather than returning False and then having to check the return value outside, why not just return `soup` unchanged?